website/integrations: add Mautic

movabo opened the pull request:
<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://docs.goauthentik.io/docs/developer-docs/#how-can-i-contribute
-->

## Details

Add [Mautic](https://mautic.org/) integration to Chat, Communication & Collaboration.

---

<details>
<summary>Not applicable (I hope)</summary>
## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)
</details>

Applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

movabo committed:
website/integrations: add Mautic

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,120 @@
+
+## What is Mautic
+
+> The World's Largest Open Source Marketing Automation Product
```
Could you give a better description of the product? Preferably a full/descriptive sentence

movabo replied:
I'm a little bit in a dilemma here. It seems that other services always cite a website, but I didn't find a good sentence on Mautics website or on wikipedia. Therefore, I used those bullet points.

dominic-r replied:
It should still be a/a few full sentences. Not bullet points

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,120 @@
+This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
+:::
+
+### Create a self-signed certificate
```
Under System > Certificates of the admin interface, it's possible for authentik to generate the certificate for you. Is there a reason why generating the certificate manually is preferred? From that certificate you just generated, it's possible to click the dropdown and the 2 buttons to download the certificate/private key.

movabo replied:
Oh, did not find that. However, I still added a section to check for PKCS#1 as I suppose that authentik does not guarantee the creation of PKCS#1 RSA certificates?! (Even though it did in the docker image.)

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,120 @@
+Mautic only supports RSA keys
+:::
+
+Also, Mautic requires the key to mention RSA in the header and footer. Change the private_key.pem manually if necessary:
```
> if necessary

What do you mean by this? The output won't change from one run to the other...

If you implement the previous selection, make sure to check whether the files use the PKCS#8 (`-----BEGIN PRIVATE KEY-----`)  or PKCS#1 (`-----BEGIN RSA PRIVATE KEY-----`) format

dominic-r commented on website/integrations/services/mautic/index.md:
````diff
@@ -0,0 +1,120 @@
+    ```
+    This indicates a problem with the mapping of the attributes ([Mautic configuration](#mautic-configuration) > Step 2 > Email/Username/First Name/Last Name or [authentik configuration](#authentik-configuration) > Step 2).
+- _"Unable to verify Signature"_ when logging in: The certificate does not match the key. (E.g. when the certificate was generated without the `RSA`.)
+- _"Assertions must be signed"_ when logging in: Are the checkboxes mentioned in [authentik configuration](#authentik-configuration) > Step 3 checked?
````
Could you add a "Configuration verification" section? https://github.com/goauthentik/authentik/blob/main/website/integrations/template/service.md?plain=1#L40

dominic-r commented on website/integrations/services/mautic/index.md:
````diff
@@ -0,0 +1,120 @@
+            ```
+3. Create an [SAML provider](../../../docs/add-secure-apps/providers/saml) with the following values:
+    - **ACS URL**: <kbd>https://<em>mautic.company</em>/s/saml/login_check</kbd>
+    - **Issuer**: <kbd><em>mautic.company</em></kbd> (without `https://`)
````
Since you specify the value to copy in a codeblock, you don't need to tell the user to not include the protocol

dominic-r reviewed (changes requested):
Hi, thank you for opening this PR. Here are a few comments to get this review started. I'll come back with a more in-dept review once some little internal details are sorted out. Please do let me know if you have any questions 🚀

movabo committed:
website/integrations: improve Mautic as suggested in goauthentik/authentik#13324

movabo committed:
website/integrations: change Mautic description to their websites meta-description

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+Mautic and authentik both require an X.509 certificates.
+Mautic requires the key to be in PKCS#1-format specifically (and thus RSA).
+
+To avoid changing certificates in authentik (e.g., if they are present in PKCS#8), we generate a new one:
```
```suggestion
To avoid changing certificates in authentik (e.g., if they are present in PKCS#8), go to the authentik Admin interface and generate a new one:
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+
+To avoid changing certificates in authentik (e.g., if they are present in PKCS#8), we generate a new one:
+
+1. Go to certificates and click on **Generate** and use the following values:
```
```suggestion
1. Go to **System -> Certificates** and click on **Generate**. Use the following values:
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+    - **Common Name**: `Mautic Self-signed Certificate`
+    - **Private key Algorithm**: `RSA`
+2. Click on the drop-down arrow left of the newly generated certificate and **Download certificate** to `certificate.pem` and **Download Private key** to `private_key.pem`.
+3. Make sure that the `private_key.pem` is in PKCS#1 format. Therefore, open it in a text editor and check that the header footer contain `RSA`:
```
```suggestion
3. Make sure that the `private_key.pem` is in PKCS#1 format. To verify, open the file in a text editor and check that the header footer contain `RSA`:
```

tanberry commented on website/integrations/services/mautic/index.md:
````diff
@@ -0,0 +1,127 @@
+    openssl req -new -key private_key.pem -out request.csr
+    ```
+
+    This will prompt to enter values for the certificate which you can choose freely. For some, you can use authentiks generated values:
````
```suggestion
    This will prompt to enter values for the certificate which you can choose freely. For some, you can use authentik's generated values:
```

tanberry commented on website/integrations/services/mautic/index.md:
````diff
@@ -0,0 +1,127 @@
+    openssl x509 -req -days 365 -in request.csr -signkey private_key.pem -out certificate_new.pem
+    ```
+
+6. Update the previously generated certificate in authentik:
````
How, what are the stps... please type the steps out.

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+
+## authentik configuration
+
+1. Because Mautic requires a first name and last name attribute, create two [SAML provider property mappings](../../../docs/users-sources/sources/property-mappings):
```
Thank you for linking to the Docs!

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+## authentik configuration
+
+1. Because Mautic requires a first name and last name attribute, create two [SAML provider property mappings](../../../docs/users-sources/sources/property-mappings):
+    1. The first name mapping returns everything until the first space (or an empty string if there is no space):
```
```suggestion
    1. The first name mapping returns everything up to the first space (or an empty string if there is no space):
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+
+## Mautic configuration
+
+1. When running behind an SSL-terminating reverse proxy (e.g. traefik): In Configuration > System Settings, make sure that
```
```suggestion
1. When running behind an SSL-terminating reverse proxy (e.g. traefik): In **Configuration > System Settings**, make sure that:
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+1. When running behind an SSL-terminating reverse proxy (e.g. traefik): In Configuration > System Settings, make sure that
+    - the **Site URL** starts with `https://`
+    - **trusted proxies** includes the IP-address of the reverse proxy
+2. In Configuration > User/Authentication Settings, set the following values:
```
```suggestion
2. In **Configuration > User/Authentication Settings**, set the following values:
```

tanberry reviewed (changes requested):
Thanks so much for this contribution @movabo! A couple of edits/questions, then if the points that @dominic-r made have been addressed, let's get this merged!

movabo commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+1. Go to certificates and click on **Generate** and use the following values:
+    - **Common Name**: `Mautic Self-signed Certificate`
+    - **Private key Algorithm**: `RSA`
+2. Click on the drop-down arrow left of the newly generated certificate and **Download certificate** to `certificate.pem` and **Download Private key** to `private_key.pem`.
```
Am I correct by assuming that you mean caret instead of carrot?

tanberry replied:
Welp yep maybe I was hungry at the moment! Thank you!

movabo commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+    - **Organizational Unit Name**: `Self-signed`
+    - **Common Name**: `Mautic Self-signed Certificate`
+
+    Next, regenerate the certificate:
```
Yes, the next step is to generate the actual certificate that will replace the one generated by authentik.
Maybe `Next, generate the certificate with the (now) PKCS#1-compliant key and the generated signing request:` would clear that up?

movabo committed:
website/integrations: clear up Mautic instructions, improve formatting

movabo commented:
Thank you for your feedback! I addressed all points of you both, @tanberry and @dominic-r . I don't know whether you prefer to resolve discussions by yourself or if I should resolve them, that's why I didn't (e.g. if you prefer looking over changes once again).
Otherwise, I incorporated the suggestions and tried to clear the text up a little.

tanberry commented on website/integrations/services/mautic/index.md:
````diff
@@ -0,0 +1,127 @@
+    ```
+    If the key contains `RSA` in header and footer, the certificate preparation is complete and you can skip steps 4 to 6.
+4. Add `RSA` after `BEGIN`/`END` in `private_key.pem` as shown in step 3 and save the file as `private_key_new.pem`.
+5. Regenerate the X.509-certificate by first creating a signing request:
````
```suggestion
5. Regenerate the X.509-certificate by first creating a signing request, using the following command:
```

tanberry commented on website/integrations/services/mautic/index.md:
````diff
@@ -0,0 +1,127 @@
+    openssl req -new -key private_key.pem -out request.csr
+    ```
+
+    This will prompt to enter values for the certificate which you can choose freely. For some, you can use authentik's generated values:
````
```suggestion
    This will prompt you to enter values for the certificate which you can choose freely. For some, you can use authentik's generated values:
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+    - **Organizational Unit Name**: `Self-signed`
+    - **Common Name**: `Mautic Self-signed Certificate`
+
+    Next, generate the certificate with the (now) PKCS#1-compliant key and the previously generated signing request:
```
```suggestion
    5. Next, generate the certificate with the (now) PKCS#1-compliant key and the previously generated signing request, using the following command:
```

tanberry reviewed (changes requested):
Hi @movabo I finally figured out what was feeling unclear to me in the "generate and regenerate" section. My brain needed to connect the step to the command beneath it. Please take a look...

movabo committed:
website/integrations: clear up Mautic instructions for certificate generation

movabo commented:
Ah, okay. I think you suggestions (which I added just now) prevent that from now on.

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+### Prepare the certificates
+
+Mautic and authentik both require an X.509 certificates.
+Mautic requires the key to be in PKCS#1-format specifically (and thus RSA).
```
```suggestion
Mautic and authentik both require X.509 certificates. However, Mautic specifically requires the key to be in PKCS#1 format (which means it must be RSA).
```

This and previous line which I can't select with this dialog on mobile

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+
+To avoid changing certificates in authentik (e.g., if they are present in PKCS#8), go to the authentik Admin interface and generate a new one:
+
+1. Go to **System -> Certificates** and click on **Generate**. Use the following values: - **Common Name**: `Mautic Self-signed Certificate`
```
Do we need to hardcode the name?

movabo replied:
No, not necessarily, but in my opinion it makes the (quite complicated) instructions a bit more readable.

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+
+1. Go to **System -> Certificates** and click on **Generate**. Use the following values: - **Common Name**: `Mautic Self-signed Certificate`
+    - **Private key Algorithm**: `RSA`
+2. Click on the caret (**>**) to the left of the newly generated certificate and click both **Download certificate** (downloads a file named `certificate.pem`) and **Download Private key** (downloads a file named `private_key.pem`).
```
```suggestion
2. Click the caret (**>**) next to the newly generated certificate, then select **Download certificate** to get the `certificate.pem` file and **Download Private key** to get the `private_key.pem` file.
```

If this makes sense

movabo replied:
Yes, I think it's even better because the file names are not exactly `private_key.pem` but only end as such.

dominic-r replied:
......

dominic-r replied:
i give up

dominic-r replied:
can we please make this entire page technically accurate before proceeding

movabo replied:
Well, it's quite hard to find a balance with all those steps involved. Here, the problem is that the Common Name in the previous step (where you asked whether it needs to be hardcoded) directly affects the names of the downloaded certificate/key.
That's why my [original suggestion](https://github.com/goauthentik/authentik/pull/13324/commits/5fd18ba30835a1e396e585fc1d02e855df5a2249#diff-1ddc1bd4786d5f8f051a2dac23e753fe67a41212b3b87c681010adefdab4de13R41) was
> **Download certificate** to `certificate.pem` and **Download Private key** to `private_key.pem`

dominic-r replied:
The original wording seems to indicate that the user would need to rename the file while downloading it, which is not offered in all browsers/at all times (for some reason or other). For now, let's hardcode a common name and get the .pem file names exact for the future?

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+1. Go to **System -> Certificates** and click on **Generate**. Use the following values: - **Common Name**: `Mautic Self-signed Certificate`
+    - **Private key Algorithm**: `RSA`
+2. Click on the caret (**>**) to the left of the newly generated certificate and click both **Download certificate** (downloads a file named `certificate.pem`) and **Download Private key** (downloads a file named `private_key.pem`).
+3. Make sure that the `private_key.pem` is in PKCS#1 format. To verify, open the file in a text editor and check that the header footer contain `RSA`:
```
A very rough draft of what I'd potentially say

---

3. Make sure that the `private_key.pem` is in PKCS#1 format. To verify, use `grep` to check for `RSA` in the header and footer of the file:
    ```sh
    grep "RSA PRIVATE KEY" private_key.pem
    ```
    If the command returns the correct match (e.g., `-----BEGIN RSA PRIVATE KEY-----`), the key is in PKCS#1 format, and you can skip steps 4 to 7.

4. If the key is not in PKCS#1 format, add `RSA` after `BEGIN`/`END` in `private_key.pem` as shown below and save the file as `private_key_new.pem`:
    ```diff
    - -----BEGIN PRIVATE KEY-----
    + -----BEGIN RSA PRIVATE KEY-----
    ```
    and
    ```diff
    - -----END PRIVATE KEY-----
    + -----END RSA PRIVATE KEY-----
    ```

5. Regenerate the X.509 certificate by first creating a signing request using the following command:
    ```sh
    openssl req -new -key private_key.pem -out request.csr
    ```
    This will prompt you to enter values for the certificate, which you can choose freely. For some fields, you can use authentik's generated values:
    - **Organization Name**: `authentik`
    - **Organizational Unit Name**: `Self-signed`
    - **Common Name**: `Mautic Self-signed Certificate`

6. Next, generate the certificate with the (now) PKCS#1-compliant key and the previously generated signing request using the following command:
    ```sh
    openssl x509 -req -days 365 -in request.csr -signkey private_key.pem -out certificate_new.pem
    ```

7. In authentik, navigate to **System > Certificates** and click **Edit** on the previously generated certificate. Click on the description below the text inputs to activate the inputs.
    - **Certificate**: Enter the contents of `certificate_new.pem`
    - **Private Key**: Enter the contents of `private_key_new.pem`
    - Click **Update**

dominic-r replied:
Tho, again, it would be best to get a 100% reproducible method, preferably directly in authentik, which would guarantee the correct key format. Also, if you regenerate the certificate in soon-to-be step 5, have you ensured everything still works fine later on? I don't know that much cryptography wise, this might be a stupid question

movabo replied:
As per my understanding it should all work the same.
For authentik, I couldn't find an option to force the generation to be PKCS#1 compliant and Mautic (or more specific the framework Mautic uses) explicitly checks for those headers.
While the current (docker) version of authentik actually generates PKCS#1 certificates, I guess that future versions of authentik won't be guaranteed to do that as well?

dominic-r replied:
I think it's best to assume we will continue generating PKCS#1 certs for now

dominic-r replied:
Format doesn't seem to have been changed since the feature was first introduced (5 years). It will most certainly stay the same

dominic-r reviewed (changes requested):
Hi @movabo . Here's my little review for the authentik configuration. Going a little further than what Tana said as it's the first time we actually have to include these steps in our docs. Please do LMK if you have any questions/need clarifications

movabo commented:
> as it's the first time we actually have to include these steps in our docs

Ah, okay, I was already wondering if my instructions were so unclear because other integration PR's seemed to just go trough. It was quite a hassle to figure it all out, that's why I created the PR so someone in the future will hopefully have it a little easier. :)

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+    - **Private Key**: Enter the contents of `private_key_new.pem`
+    - Click on **Update**
+
+## authentik configuration
```
The certificate mess from above ^^ should be included under the authentik configuration section. Could you also update this to be in line with the new template? https://github.com/goauthentik/authentik/pull/12929 should be merged to main soon, but it's getting blocked by something else. Here's an example from another PR which has property mapping configuration: https://github.com/goauthentik/authentik/pull/13144/files#diff-e5710672bf55030bb3352ed289b1acc49be1a41038ab3f2ceee66da1d9d1d713R29 Thanks

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+
+## Mautic configuration
+
+1. When running behind an SSL-terminating reverse proxy (e.g. traefik): In **Configuration > System Settings**, make sure that:
```
This could be added as a :::note instead as the steps below should be more on configuring SAML

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+    - the **Site URL** starts with `https://`
+    - **trusted proxies** includes the IP-address of the reverse proxy
+2. In **Configuration > User/Authentication Settings**, set the following values:
+    - **Entity ID for the IDP**: <kbd>https://<em>mautic.company</em></kbd>
```
Knowing that Entity ID = Issuer, shouldn't the value of the issuer on line 91 be identical to this one?

movabo replied:
As per the [mautic docs](https://docs.mautic.org/en/5.x/authentication/authentication.html), it should be as described.

dominic-r replied:
We can make it whatever we want, right? To prevent confusion, shouldn't we set it to `authentik` or similar

movabo replied:
Mautic does not allow distinct configurations of Entity ID and Issuer.
In the docs, they just mentioned the one to be with `https://` and the other one to be without.
I don't know if it's hardcoded in their system (or even affects their system at all, because I'm not that deep in the authentication subject either).
For me, the values from their documentation worked. As per "never change a running system," I didn't further investigate.

dominic-r commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,127 @@
+
+## Configuration verification
+
+To confirm that authentik is properly configured with Mautic, open a new incognito/private window or another browser and
```
Could you remove the new lines in the middle of sentences?

dominic-r commented:
BTW, thank you for bearing with us  🤣

movabo committed:
website/integrations: move Mautic certificate generation to Troubleshooting

tanberry reviewed (approved):
Thank you so much @movabo for this contribution, and for your patience with our many edits and questions and adjustments. We are approving it, and will merge after all build checks pass. The one remaining question I have is about the Troubleshooting section... are the Danger alerts truly a "Danger" level? Will data be lost, systems crashed, etc.? Or will things simply not work? If nothing crashes and burns, lets use a warning level instead of danger.

movabo commented:
> Will data be lost, systems crashed, etc.?

No.

However, I used the alerts for displaying the error messages Unfortunately docusaurus does not have (or I did not find) the possibility to change the wording from "danger" to "error."
I found it to be the best fit for making different errors distinguishable. I also tested only the horizontal rulers, but found it harder to differentiate the errors. And on the other hand, warning would suggest that it's only a warning message being displayed to the user.
Also, I did not add a subsection per error because I would use the error-message as headlines of some are quite long.

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+## Troubleshooting
+
+:::danger
```
```suggestion
### `Uncaught PHP Exception TypeError`
```

tanberry commented on website/integrations/services/mautic/index.md:
````diff
@@ -0,0 +1,211 @@
+mautic.CRITICAL: Uncaught PHP Exception TypeError: "Mautic\UserBundle\Entity\User::getUserIdentifier(): Return value must be of type string, null returned" at /app/bundles/UserBundle/Entity/User.php line 335 {"exception":"[object] (TypeError(code: 0): Mautic\\UserBundle\\Entity\\User::getUserIdentifier(): Return value must be of type string, null returned at /app/bundles/UserBundle/Entity/User.ph ...
+```
+
+:::
````
```suggestion
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+<hr />
+
+:::danger
```
```suggestion
### `Unable to verify Signature`
```

tanberry commented on website/integrations/services/mautic/index.md:
````diff
@@ -0,0 +1,211 @@
+:::danger
+
+```
+mautic.CRITICAL: Uncaught PHP Exception TypeError: "Mautic\UserBundle\Entity\User::getUserIdentifier(): Return value must be of type string, null returned" at /app/bundles/UserBundle/Entity/User.php line 335 {"exception":"[object] (TypeError(code: 0): Mautic\\UserBundle\\Entity\\User::getUserIdentifier(): Return value must be of type string, null returned at /app/bundles/UserBundle/Entity/User.ph ...
````
```suggestion
```
> mautic.CRITICAL: Uncaught PHP Exception TypeError: "Mautic\UserBundle\Entity\User::getUserIdentifier(): Return value must be of type string, null returned" at /app/bundles/UserBundle/Entity/User.php line 335 {"exception":"[object] (TypeError(code: 0): Mautic\\UserBundle\\Entity\\User::getUserIdentifier(): Return value must be of type string, null returned at /app/bundles/UserBundle/Entity/User.php:335)"}
```
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+:::danger
+
+> Unable to verify Signature
```
```suggestion
> `Unable to verify Signature`
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+:::
+
+when logging in indicates that the certificate does not match the private key.
```
```suggestion
This error occurs when logging in indicates that the certificate does not match the private key.
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+<hr />
+
+:::danger
```
```suggestion
### Assertions
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+:::danger
+
+> Assertions must be signed
```
```suggestion
> `Assertions must be signed`
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+> Assertions must be signed
+
+:::
```
```suggestion
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+<hr />
+
+:::danger
```
```suggestion
### Invalid private key
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+:::danger
+
+> Private key is invalid. It should begin with `-----BEGIN RSA PRIVATE KEY-----` or `-----BEGIN ENCRYPTED PRIVATE KEY-----`
```
```suggestion
```
> Private key is invalid. It should begin with `-----BEGIN RSA PRIVATE KEY-----` or `-----BEGIN ENCRYPTED PRIVATE KEY-----`
```
```

tanberry commented on website/integrations/services/mautic/index.md:
```diff
@@ -0,0 +1,211 @@
+
+> Private key is invalid. It should begin with `-----BEGIN RSA PRIVATE KEY-----` or `-----BEGIN ENCRYPTED PRIVATE KEY-----`
+
+:::
```
```suggestion
```

tanberry commented:
Hi @Mautic, did you want to look at the formatting changes I made to the Troubleshooting section, or is it ok with you if I commit those changes and then go ahead and merge your PR?

movabo committed:
website/integrations: Improve Mautic formatting

movabo commented:
Thank you for the reminder. I improved the formatting as per your changes and to further match the formatting of the error messages in mautic itself (e.g. the "Invalid private key" message is not as code).
One last thing: Shouldn't `Sign assertions` and `Sign responses` be bold instead of code? They are labels for check/toggle boxes.

tanberry commented:
> Thank you for the reminder. I improved the formatting as per your changes and to further match the formatting of the error messages in mautic itself (e.g. the "Invalid private key" message is not as code). One last thing: Shouldn't `Sign assertions` and `Sign responses` be bold instead of code? They are labels for check/toggle boxes.

Great, thanks so much. And Yes, correct, any labels should be bold font, code code. Thanks!

movabo committed:
website/integrations: For Mautic, make labels bold

movabo commented:
Okay, changed that now as well. I hope now everything is fine! :)

tanberry commented:
> Okay, changed that now as well. I hope now everything is fine! :)

Thank you again @movabo both for this contribution and your patience with our edits. ;-)
